### PR TITLE
Fix parameter_size in APT::GetStartupArgument

### PR DESCRIPTION
**Pull request by B3n30**
3dbrew says it's capped to 0x300, so silently set it to 0x300 and let the code continue

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3921)

### CODE REVIEW

File: src/core/hle/service/apt/apt.cpp [hunk 1]
```diff
@@ -620,12 +620,14 @@ void Module::Interface::GetStartupArgument(Kernel::HLERequestContext& ctx) {
...
-            parameter_size);
-        return;
+    const u32 max_parameter_size{0x1000};
+
+    if (parameter_size > max_parameter_size) {
+        parameter_size = max_parameter_size;
+        LOG_ERROR(Service_APT,
+                  "Parameter size is outside the valid range (capped to {:#010X}): "
+                  "parameter_size={:#010X}",
+                  max_parameter_size, parameter_size);
```
Comment (wwylele) [inline src/core/hle/service/apt/apt.cpp]: Sorry, just realized this problem... you log `parameter_size` after modifying it, so it will be always 0x300

Comment (B3n30) [reply src/core/hle/service/apt/apt.cpp]: fixed

**Comment by baka0815** [comment]
@B3n30 you say you cap at ``0x300`` but the code caps at ``0x1000`` - is this intentional?

**Comment by wwylele** [comment]
@baka0815 the code was updated after RE the system module. He just forgot to edit the description.

**Comment by tywald** [comment]
```
wwylele / 白疾風 - Last Saturday at 6:46 PM
@B3N30 I just checked the APT code. It is indeed a silent min. However, the capped size is 0x1000
B3N30 - Last Saturday at 6:46 PM
LOL, what?
wwylele / 白疾風 - Last Saturday at 6:46 PM
They might changed size at some point during system update
my APT dump is from a 9.3 or a 11.4 system. I don't remember
B3N30 - Last Saturday at 6:47 PM
so, 0x1000 it is :thonking:
```
From Discord.

**Comment by baka0815** [comment]
@wwylele, @tywald thanks for the clarification

**Pull request merged**